org.jcryptool.visual.rabin/src/org/jcryptool/visual/rabin/GUIHandler.java

```diff
@@ -0,0 +1,1497 @@
+			
+			if(!checkUpperLimit) {
+				if(!(ptmp.compareTo(limitUpPQ) <= 0) && !(qtmp.compareTo(limitUpPQ) <= 0)) {
+					cmbP.setBackground(colorBackgroundWrong);
+					cmbQ.setBackground(colorBackgroundWrong);
+					pWarning.setText(MessageFormat.format(strUpperLimitRestriction, limitExpPQ));
+					qWarning.setText(MessageFormat.format(strUpperLimitRestriction, limitExpPQ));
+					showControl(pWarning);
+					showControl(qWarning);
+				}
+				
+				if(!(ptmp.compareTo(limitUpPQ) <= 0) && (qtmp.compareTo(limitUpPQ) <= 0)) {
+					cmbP.setBackground(colorBackgroundWrong);
+					cmbQ.setBackground(colorBackgroundCorrect);
+					pWarning.setText(MessageFormat.format(strUpperLimitRestriction, limitExpPQ));
+					showControl(pWarning);
+				}
+				
+				if((ptmp.compareTo(limitUpPQ) <= 0) && !(qtmp.compareTo(limitUpPQ) <= 0)) {
+					cmbP.setBackground(colorBackgroundCorrect);
+					cmbQ.setBackground(colorBackgroundWrong);
+					qWarning.setText(MessageFormat.format(strUpperLimitRestriction, limitExpPQ));
+					showControl(qWarning);
+				}
+				return;
+			}
+			
+			
+			if(rabinFirst.isSuitablePrime(ptmp) && rabinFirst.isSuitablePrime(qtmp)) {
+				
+				if(!ptmp.equals(qtmp)) {
+					if(rabinFirst.isCompositeSuitable(ptmp, qtmp)) {
+						cmbP.setBackground(colorBackgroundCorrect);
+						cmbQ.setBackground(colorBackgroundCorrect);
+						hideControl(pWarning);
+						hideControl(qWarning);
+						hideControl(nWarning);
+						if(rftc.btnGenKeysMan.getSelection()) {
+							rftc.btnStartGenKeys.setEnabled(true);
+						}
+					}
+					else {
+						cmbP.setBackground(colorBackgroundWrong);
+						cmbQ.setBackground(colorBackgroundWrong);
+						nWarning.setText(strNotSuitableN);
+						showControl(nWarning);
+					}
+				}
+				else {
+					cmbP.setBackground(colorBackgroundWrong);
+					cmbQ.setBackground(colorBackgroundWrong);
+					nWarning.setText(strPnotEqualQ);
+					showControl(nWarning);
+				}
+			}
+			else if(rabinFirst.isSuitablePrime(ptmp) && !rabinFirst.isSuitablePrime(qtmp)) {
+				
+				cmbP.setBackground(colorBackgroundCorrect);
+				cmbQ.setBackground(colorBackgroundWrong);
+				qWarning.setText(strQnotSuitable);
+				showControl(qWarning);
+			}
+			else if(!rabinFirst.isSuitablePrime(ptmp) && rabinFirst.isSuitablePrime(qtmp)) {
+				
+				cmbP.setBackground(colorBackgroundWrong);
+				cmbQ.setBackground(colorBackgroundCorrect);
+				pWarning.setText(strPnotSuitable);
+				showControl(pWarning);
+			}
+			else if(!rabinFirst.isSuitablePrime(ptmp) && !rabinFirst.isSuitablePrime(qtmp)) {
+				cmbP.setBackground(colorBackgroundWrong);
+				cmbQ.setBackground(colorBackgroundWrong);
+				pWarning.setText(strPnotSuitable);
+				qWarning.setText(strQnotSuitable);
+				showControl(pWarning);
+				showControl(qWarning);
+			}
+		}	
+	}
+	
+	
+	
+	
+	/**
+	 * @param elem
+	 * @param a
+	 * @return startIdx the startIdx of elem in a
+	 */
+	public int getStartIdx(int elem, ArrayList<String> a) {
```

Comment:
In our company we often use reduced docstrings, when the parameters are pretty obvious, and one sentence is more or less enough.

```suggestion
	/** Find the start index by summing up all string lengths until (exclusive) limit plus separator length. */
	public int getStartIdx(int limit, ArrayList<String> strings) {
```